javalib/src/main/scala/java/util/Properties.scala

```diff
@@ -0,0 +1,64 @@
+package java.util
+
+import java.{util => ju}
+
+import scala.collection.JavaConversions
+import scala.collection.JavaConversions._
+
+class Properties(protected val defaults: Properties)
+    extends ju.Hashtable[AnyRef, AnyRef] {
+
+  def this() = this(null)
+
+  def setProperty(key: String, value: String): AnyRef =
+    put(key, value)
+
+  // def load(reader: Reader): Unit
+  // def load(inStream: InputStream): Unit
+  // @deprecated("", "") def save(out: OutputStream, comments: String): Unit
+  // def store(writer: Writer, comments: String): Unit
+  // def store(out: OutputStream, comments: String): Unit
+  // def loadFromXML(in: InputStream): Unit
+  // def storeToXML(os: OutputStream, comment: String): Unit
+  // def storeToXML(os: OutputStream, comment: String, encoding: String): Unit
+
+  def getProperty(key: String): String = {
+    get(key) match {
+      case value: String => value
+
+      case _ =>
+        if (defaults != null) defaults.getProperty(key)
+        else null
+    }
+  }
+
+  def getProperty(key: String, defaultValue: String): String = {
+    getProperty(key) match {
+      case null  => defaultValue
+      case value => value
+    }
+  }
+
+  def propertyNames(): ju.Enumeration[_] = {
+    val thisSet = keySet()
+    val defaultsIterator =
+      if (defaults != null) defaults.propertyNames().toIterator
+      else scala.collection.Iterator.empty
+    val filteredDefaults = defaultsIterator.collect {
+      case k: AnyRef if !thisSet(k) => k
+    }
+    JavaConversions.asJavaEnumeration(thisSet.iterator() ++ filteredDefaults)
+  }
+
+  def stringPropertyNames(): ju.Set[String] = {
+    val set = new ju.HashSet[String]
+    propertyNames().foreach {
+      case key: String => set.add(key)
+      case _           => // Ignore key
+    }
+    ju.Collections.synchronizedSet(set)
```

Comment:
I would follow the spec, here. It doesn't say anything about synchronization. Since we don't need synchronizing anyway, I would just return `set` and avoid depending on all the code that hides behind `synchronizedSet()`.